benches/performance.rs

```diff
@@ -0,0 +1,456 @@
+use criterion::{Criterion, black_box, criterion_group, criterion_main};
+use multi_index_map::MultiIndexMap;
+
+
+#[derive(Hash, PartialEq, Eq, Clone, Debug)]
+struct TestNonPrimitiveType(u32);
+
+#[derive(MultiIndexMap, Debug, Clone)]
+pub struct TestElementWithOnlyIndexedFields {
+    #[multi_index(hashed_unique)]
+    field_hashed_unique: TestNonPrimitiveType,
+    #[multi_index(hashed_non_unique)]
+    field_hashed_non_unique: TestNonPrimitiveType,
+    #[multi_index(ordered_unique)]
+    field_ordered_unique: u32,
+    #[multi_index(ordered_non_unique)]
+    field_ordered_non_unique: u32,
+}
+
+const BENCH_SIZES: &[u32] = &[100u32, 1000u32, 10000u32, 100000u32];
+
+fn insert_benchmark(c: &mut Criterion) {
+    fn inner(c: &mut Criterion, n: u32) {
+        let mut map = black_box(MultiIndexTestElementWithOnlyIndexedFieldsMap::default());
+        c.bench_function(&format!("insert_bench_{}", n), |b|{b.iter(|| {
```

Comment:
Please do run rustfmt, I notice a few places in where the formatting is a bit off

Reply:
[2bff67e](https://github.com/lun3x/multi_index_map/pull/23/commits/2bff67eec2587697fc6088cd8e6a6bd722e992e2)

Reply:
[2bff67e](https://github.com/lun3x/multi_index_map/pull/23/commits/2bff67eec2587697fc6088cd8e6a6bd722e992e2)